### PR TITLE
Add tests for simulated variant sets

### PR DESCRIPTION
**Pull request by dcolligan**
Increase required test coverage to 85%

Issue #222
Issue #195

### CODE REVIEW

File: tests/unit/test_simulator.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+class TestSimulatedVariantSet(unittest.TestCase):
+    """
+    Test properties of the SimulatedVariantSet
+    """
+    def setUp(self):
+        self.randomSeed = 0
+        self.numCalls = 1
+        # ensure variantDensity is >= 1 so we get deterministic behavoir
+        self.variantDensity = 2
```
Comment (jeromekelleher) [inline tests/unit/test_simulator.py]: Just set this to 1, it should be interpreted as a probability.

File: tests/unit/test_simulator.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        self.callSetIds = ['unused']
+        self.bases = ["A", "C", "G", "T"]
+
+    def _getSimulatedVariantSet(self):
+        simulatedVariantSet = variants.SimulatedVariantSet(
+            self.randomSeed, self.numCalls,
+            self.variantDensity, self.variantSetId)
+        return simulatedVariantSet
+
+    def _getSimulatedVariantsList(self, simulatedVariantSet=None):
```
Comment (jeromekelleher) [inline tests/unit/test_simulator.py]: Why do we need the parameter here? It looks to me like it's not actually used.

**Comment by jeromekelleher** [comment]
+1 modulo niggles.

File: tests/unit/test_simulator.py (hunk 2 above)
Comment (dcolligan) [inline tests/unit/test_simulator.py]: see `testConsistency`

**Comment by dcolligan** [comment]
@jeromekelleher pushed commit with your suggestions

File: tests/unit/test_simulator.py (hunk 2 above)
Comment (jeromekelleher) [inline tests/unit/test_simulator.py]: Nope, still not seeing where this parameter is used. It doesn't have any effect anyway, even if it is used: you don't use `simulatedVariantSet` anywhere in the function.

Comment (dcolligan) [inline tests/unit/test_simulator.py]: ah, well that's a bug in the code!

**Comment by dcolligan** [comment]
@jeromekelleher pushed a commit to fix that bug

File: tests/unit/test_simulator.py (hunk 2 above)
Comment (dcolligan) [inline tests/unit/test_simulator.py]: and check lines 84-85 ... that's where the parameter is passed

**Comment by jeromekelleher** [comment]
Ah, now I get it.

+1.

**Pull request merged**